packages/cyberstorm-forms/src/components/FormSelect.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import { z, ZodObject } from "zod";
+import { ZodRawShape } from "zod/lib/types";
+import { Path, useController } from "react-hook-form";
+import styles from "./FormSelect.module.css";
+import * as Select from "@radix-ui/react-select";
+import React, { PropsWithChildren } from "react";
+import { Icon } from "@thunderstore/cyberstorm";
+import { faCaretDown, faCheckCircle } from "@fortawesome/pro-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+export type FormSelectProps<
+  Schema extends ZodObject<Z>,
+  Z extends ZodRawShape
+> = {
+  // The schema is required to allow TS to infer valid values for the name field
+  schema: Schema;
+  name: Path<z.infer<Schema>>;
+  options: { value: string; label: string }[];
```

Comment:
Nice to see you've chosen to use the value/label approach instead of a single string this time around. Although it does make the `SelectSear` stick out like a sore thumb.